### PR TITLE
Add "default pull action" config in setting dialog

### LINKED ISSUE #6443: [Accessibility] "Default pull action" configuration
## Feature description

The current configuration for the default pull action is confined to the dropdown menu under a toolstrip button in the main toolbar:
image

This setting is hard to discover especially for those users who invokes the functionality via the Windows Explorer shell extensions or the VSIX.

The configuration must be available via the main Settings dialog.
The suggested placement is on the "Advanced" tab after the "Commit" section.

Relates to #6194 

## Environment

- Git Extensions 3.0.2.5232

### PR DESCRIPTION
**Pull request by FisherTsai**
Fixes #6443 

## Proposed changes

- Add "default pull action" config in setting dialog

### Before
before

### After
after

## Test methodology 

- Use the new combo box to set a pull action and back to the main window to check if the pull button on toolbar shows the correct action icon. And then click the button to see if the action we set is triggered.

## Test environment(s) 

- GIT 2.22.0.windows.1
- Windows 10 build 18362.239

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by gerhardol** [comment]
This dialog should be in _Detailed /Browse repository window_
(The Commit snippet should be moved too...)

I suggest that the dropdown in the menu is replaced with a shortcut to the settings.

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -1459,6 +1459,8 @@ private void OnShowSettingsClick(object sender, EventArgs e)
             _dashboard?.RefreshContent();
 
             _gitStatusMonitor.Active = NeedsGitStatusMonitor() && Module.IsValidGitWorkingDir();
+
+            RefreshDefaultPullAction();
```
[the remark this answers is not in the document]
Comment (FisherTsai) [reply GitUI/CommandsDialogs/FormBrowse.cs]: To update pull action of the pull button on toolbar after user leaves setting dialog. That button doesn't show the new action we choose without calling this function here.

**Comment by gerhardol** [comment]
> However it is more a general settings, and is more appropriately placed at the "General" tab under the [Behaviour] group.

I disagree:
> This dialog should be in Detailed /Browse repository window
(The Commit snippet should be moved too...)

**Comment by RussKie** [comment]
The way I perceive it, the pull _behaviour_ is how the app works in _general_, and it isn't tied to the way a repo is browsed.
The pull functionality can be invoked from the Windows shell extension and from the VS extension, neither of those provide browsing functionality.

**Comment by gerhardol** [comment]
> The way I perceive it, the pull behaviour is how the app works in general, and it isn't tied to the way a repo is browsed.

I can agree with this reasoning, but the General tab will not fit this information and the size should not be increased. Do you want a separate page below General?

**Comment by RussKie** [comment]
I think we have sufficient space on the General tab to fit another dropdown
image

**Comment by gerhardol** [comment]
> I think we have sufficient space on the General tab to fit another dropdown

The size of Settings is already on the limit for some laptops
That would use the last available spot
(I rather keep Advanced then move to General.)

**Comment by RussKie** [comment]
The screenshot was taken at the smaller possible size of the dialog. So there is space to accommodate the dropdown

**Comment by RussKie** [comment]
@FisherTsai please move the setting to the General tab in Behaviour section under the "Default clone destination".
thank you

**Comment by FisherTsai** [comment]
> 
> 
> @FisherTsai please move the setting to the General tab in Behaviour section under the "Default clone destination".
> thank you

Here is the screenshot for the latest modification.

**Comment by pmiossec** [comment]
This is an important setting.
I tend to think that we should put it first in the 'behavior' group.

What do you think?

**Comment by RussKie** [comment]
I don't think the order within the group is terribly important, arguably all and each setting can be argued as important.
To me it fits right in with another "default" setting.

**Comment by RussKie** [comment]
Thank you

**Pull request merged**